### DISCUSSION TITLE
How to use vcpkg in Android studio?

### DISCUSSION THREAD

**Discussion by PeacefulWindy** [microsoft/vcpkg#43029] [Q&A] [answered]
I use this CMakeLists.txt in my project.

set(CMAKE_TOOLCHAIN_FILE ${CMAKE_CURRENT_SOURCE_DIR}/vcpkg/scripts/buildsystems/vcpkg.cmake)

```
if(ANDROID)
    if (ANDROID_ABI MATCHES "arm64-v8a")
        set(VCPKG_TARGET_TRIPLET "arm64-android" CACHE STRING "" FORCE)
    elseif(ANDROID_ABI MATCHES "armeabi-v7a")
        set(VCPKG_TARGET_TRIPLET "arm-android" CACHE STRING "" FORCE)
    elseif(ANDROID_ABI MATCHES "x86_64")
        set(VCPKG_TARGET_TRIPLET "x64-android" CACHE STRING "" FORCE)
    elseif(ANDROID_ABI MATCHES "x86")
        set(VCPKG_TARGET_TRIPLET "x86-android" CACHE STRING "" FORCE)
    else()
    set(VCPKG_CHAINLOAD_TOOLCHAIN_FILE $ENV{ANDROID_NDK_HOME}/build/cmake/android.toolchain.cmake)
    endif()
endif()
project(myProject)
...
```

but android studio maybe not to find vcpkg library.find_package failed.
Visual studio use CMAKE_TOOLCHAIN_FILE  can work.

And then,vcpkg export android prefab not work.
So I don't understand how to use in Android Studio.

**Comment by dg0yt** [comment, marked as answer]
I can't say much about android prefab and Android Studio.

In general, if not using presets or other configuration input, `ANDROID` and `ANDROID_ABI` are not defined before `project`. 
With `VCPKG_TARGET_TRIPLET` defined in advance, the vcpkg cmake toolchain file will take care of chainloading vcpkg's android toolchain, which in turn will set the variables that describe the target system.

**Reply by PeacefulWindy** [reply]
Oh yes.
I new test andoid project to test it.
Confirm ANDROID not defined before project,remove it.
Now I set VCPKG_TARGET_TRIPLET,CMAKE_TOOLCHAIN_FILE and VCPKG_CHAINLOAD_TOOLCHAIN_FILE (ANDROID_ABI is autoset from Android Studio).
And made a another mistake about vcpkg.json path,it not auto install manifest from outside path.
Finally,it can work.
Thanks.^_^
